[Tracing][PHP] Add annotations doc (#16778)
DataDog/documentation: content/en/tracing/trace_collection/custom_instrumentation/php.md

If you have not yet read the instructions for auto-instrumentation and setup, start with the <a href="/tracing/setup/php/">PHP Setup Instructions</a>. Even if Datadog does not officially support your web framework, you may not need to perform any manual instrumentation. See <a href="/tracing/setup/php/#automatic-instrumentation">automatic instrumentation</a> for more details.
</div>

## Annotations

> If you run on PHP 8, starting v0.84 of the tracer, you can add attributes to your code to instrument it. It is a lighter alternative to [custom instrumentation written in code](#writing-custom-instrumentation). For instance, just add `#[DDTrace\Trace]` attribute to methods for Datadog to trace them.

Reviewer: ```suggestion
If you are using PHP 8, as of v0.84 of the tracer, you can add attributes to your code to instrument it. It is a lighter alternative to [custom instrumentation written in code](#writing-custom-instrumentation). For example, add the `#[DDTrace\Trace]` attribute to methods for Datadog to trace them.
```

```php
<?php
class Foo {

>     #[DDTrace\Trace(name: "simplename", resource: "rsrc", type: "typeee", service: "test", tags: ["a" => "b"])]

Reviewer: is "typeee" an actual type...? if not can we make the example a little more realistic?

Author: done in 19366c5d4ecfc72675ee41b6454fd20bd0f8f274

    static function simple($arg) {}

    #[DDTrace\Trace]
    function bar() {

>       Foo::simple(1);

Reviewer: can we avoid foobars here and put in something more realistic for our readers?

Author: done in 19366c5d4ecfc72675ee41b6454fd20bd0f8f274
